liblangutil/SourceLocation.h

```diff
@@ -49,6 +49,23 @@ struct SourceLocation
 
 	bool isEmpty() const { return start == -1 && end == -1; }
 
+	/// @returns the smallest SourceLocation that contains both @param _a and @param _b.
+	static SourceLocation smallestCovering(SourceLocation const& _a, SourceLocation const& _b)
+	{
+		SourceLocation result = _a;
```

Comment:
Pass `_a` by value and move here?

Reply:
Hm... true. That's better.

Reply:
Actually if I pass ``_a`` by value I can just work on ``_a`` instead and return ``_a`` and skip the whole ``result`` thing.